### ISSUE TITLE
make 18 compatible

### ISSUE THREAD

**Issue by wiswedel** [nextcloud/social#844]
This is a featured app, but not yet made 18-ready. Please don't let that get forgotten.

cc @daita

**Comment by StCyr** [comment]
what's missing? Can you point me to some doc?

**Comment by juliushaertl** [comment]
@StCyr Most likely just a bump of the max-version in appinfo/info.xml and a release to the appstore. cc @daita

**Comment by StCyr** [comment]
Hi @juliushaertl 

I've bumped max-version to 18 and tried to enabled the app but it fails with "An error occured during the request. Unable to proceed"

In the logs I find lines like the following:

```
Jan 22 19:28:13 vps612142 Nextcloud[716]: {"reqId":"S8QWRsGI7vnv002WcyEj","level":3,"time":"2020-01-22T18:28:13+00:00","remoteAddr":"94.104.124.217","user":"admin","app":"password_policy","method":"GET","url":"/ocs/v2.php/core/navigation/apps?format=json","message":"{\"Exception\":\"RuntimeException\",\"Message\":\"App class OCA\\\\Password_Policy\\\\AppInfo\\\\Application is not setup via query() but directly\",\"Code\":0,\"Trace\":[{\"file\":\"/var/www/nextcloud/apps/password_policy/lib/AppInfo/Application.php\",\"line\":40,\"function\":\"__construct\",\"class\":\"OCP\\\\AppFramework\\\\App\",\"type\":\"->\",\"args\":[\"password_policy\"]},{\"file\":\"/var/www/nextcloud/apps/password_policy/appinfo/app.php\",\"line\":23,\"function\":\"__construct\",\"class\":\"OCA\\\\Password_Policy\\\\AppInfo\\\\Application\",\"type\":\"->\",\"args\":[]},{\"file\":\"/var/www/nextcloud/lib/private/legacy/app.php\",\"line\":266,\"args\":[\"/var/www/nextcloud/apps/password_policy/appinfo/app.php\"],\"function\":\"require_once\"},{\"file\":\"/var/www/nextcloud/lib/private/legacy/app.php\",\"line\":155,\"function\":\"requireAppFile\",\"class\":\"OC_App\",\"type\":\"::\",\"args\":[\"password_policy\"]},{\"file\":\"/var/www/nextcloud/lib/private/legacy/app.php\",\"line\":128,\"function\":\"loadApp\",\"class\":\"OC_App\",\"type\":\"::\",\"args\":[\"password_policy\"]},{\"file\":\"/var/www/nextcloud/ocs/v1.php\",\"line\":56,\"function\":\"loadApps\",\"class\":\"OC_App\",\"type\":\"::\",\"args\":[]},{\"file\":\"/var/www/nextcloud/ocs/v2.php\",\"line\":24,\"args\":[\"/var/www/nextcloud/ocs/v1.php\"],\"function\":\"require_once\"}],\"File\":\"/var/www/nextcloud/lib/public/AppFramework/App.php\",\"Line\":80,\"CustomMessage\":\"--\"}","userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.13; rv:72.0) Gecko/20100101 Firefox/72.0","version":"18.0.0.10"}
```

**Comment by juliushaertl** [comment]
The error is unrelated. But I guess there is something wrong with the database migrations, as CI also is currently unable to install the app after a clean git clone.

**Comment by jancborchardt** [comment]
@daita could you check this out? cc @violoncelloCH for reference

**Comment by loxK** [comment]
Funny, just after upgrading to 18.0.1 I got a notification recommending me to install that incompatible application.

**Comment by jancborchardt** [comment]
@daita :ping_pong:

**Comment by jancborchardt** [comment]
@StCyr @loxK or if you would like to submit a pull request bumping the max-version (now to 19 so it’s also compatible with development master) please go ahead :)

**Comment by wiswedel** [comment]
> @StCyr @loxK or if you would like to submit a pull request bumping the max-version (now to 19 so it’s also compatible with development master) please go ahead :)

Bumping the version compatibility should not be done without proper qualification if the app is _actually_ working with the server version. Otherwise the whole compatibility function is pointless.
So I see the responsibility of verifying the compatibility at the maintainer.

**Comment by StCyr** [comment]
Also, I’ve already tried to just bump this app’s max-version. And, it didn’t solved anything :(

**Comment by rubo77** [comment]
What is the problem, if you just install it with the red button anyway?

What is not working then?

**Comment by daita** [comment]
Back form holiday.

I will work on a new alpha release for NC18 within the next few weeks. Please be patient !

**Comment by 0x9060** [comment]
Any update on this? Is there a workaround for NC18?

**Comment by rubo77** [comment]
>   Is there a workaround for NC18?

yes, see https://github.com/nextcloud/social/issues/850#issuecomment-601040175

**Comment by jancborchardt** [comment]
Reopening cause it’s not 18 compatible yet. cc @daita see that quick fix above, maybe it’s helpful. :)

**Comment by rodneyrod** [comment]
Is this issue now fixed with #850 being closed?

**Comment by daita** [comment]
yes, issue can be closed, official release expected today
